app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb

```diff
@@ -3,8 +3,100 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job < ManageIQ::P
+    task = MiqTask.find(task_id)
+    task.update_message(msg)
+  end
+
+  def self.reload_job_status(job_id, options)
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    tower = playbook.manager
+    tower.with_provider_connection do |connection|
+      connection.api.jobs.find(job_id)
+    end
+  end
+
+  def self.requeue_job_status(job_id, task_id)
+    task = MiqTask.find(task_id)
+    options = task.context_data
+    job = reload_job_status(job_id, options)
+    if job.status == 'successful' || job.status == 'failed'
+      # Modify the task to include the error on fail
+      task.update_status("Finished", "Ok", "Playbook launch finished successfully") if task
+    else
+      task.update_status("Active", "Ok", "Playbook still running") if task
+      info = options[:config_info]
+      playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+      ems = playbook.manager
+      cb = {:class_name => task.class.name, :instance_id => task.id, :method_name => :queue_callback_on_exceptions, :args => ['Finished']}
+      MiqQueue.put(
+        :class_name  => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job",
+        :method_name => 'requeue_job_status',
+        :role        => 'embedded_ansible',
+        :args         => [job.id, task.id],
+        :miq_callback => cb,
+        :zone         => ems.my_zone
+      )
+    end
+  end
+
+  def self.create_and_return_inventory(tower, name, description, task_id)
+    inventory = nil
+    message = "Creating new inventory"
+    params = {:name => name, :description => description, :organization => tower.provider.default_organization }
+    tower.with_provider_connection do |connection|
+      inventory = connection.api.inventories.create!(params)['id']
+    end
+    message(message, task_id) if inventory
+    inventory
+  end
+
+  def self.build_name(basic_name)
+    "#{basic_name}_#{Time.zone.now.to_i}"
+  end
+
   def retire_now(requester = nil)
     update_attributes(:retirement_requester => requester)
     finish_retirement
   end
+
+  def self.run(options, userid = nil)
+    userid ||= 'system'
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    ems = playbook.manager
+
+    task = MiqTask.create(:name => "Running Ansible Playbook for user #{userid}", :userid => userid, :context_data => options)
+    cb = {:class_name => task.class.name, :instance_id => task.id, :method_name => :queue_callback_on_exceptions, :args => ['Finished']}
+    MiqQueue.put(
+      :class_name  => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job",
+      :method_name => 'run_playbook',
+      :role        => 'embedded_ansible',
+      :args         => [task.id],
+      :miq_callback => cb,
+      :zone         => ems.my_zone
+    )
```

Comment:
@syncrou 
Should this be returning task.id since Automate is expecting to get back a task_id